app/stac_api/validators_view.py

```diff
@@ -92,3 +92,11 @@ def validate_upload_parts(request):
         message = f'Required "parts" must be a list, not a {type(request.data["parts"])}'
         logger.error(message, extra={'request': request})
         raise ValidationError({'parts': _(message)}, code='invalid')
+
+
+def validate_renaming(serializer, id_field, original_name):
+    data = serializer.validated_data
+    if data[id_field] != original_name:
+        message = 'Renaming object is not allowed'
+        logger.error(message, extra={'request': data})
```

Comment:
`extra={'request': ...}` must be a django HttpRequest object, here data is a dictionary of the deserialized payload. In the extra it would be good to have the object name for this I suggest to pass the extra as argument and depending on which view we validate set different extra. E.g.
```python
def validate_renaming(serializer, id_field, original_name, log_extra):
   ... 
  logger.error(message, extra=log_extra)

class AssetsDetail:

def perform_upsert():
   validate_renaming(serializer, 'name', self.kwargs['asset_name'], {'asset': self.kwargs['asset_name']}
```